### PR TITLE
Introduce a RNW Fabric Sample App

### PR DESCRIPTION
**Pull request by sharath2727**
### Type of Change
Introducing a Fabric Sample App.

### Why
Showcase functionality of Fabric architecture.

### What
No changes to the codebase.

## Changelog
no

"Part of #12990"

### CODE REVIEW

File: packages/sample-app-fabric/.bundle/config
Comment (jonthysell) [inline packages/sample-app-fabric/.bundle/config:1]: You can delete this folder/file.

File: packages/sample-app-fabric/windows/SampleAppFabric.Package/Package.appxmanifest [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="SampleAppFabric"
+    Publisher="CN=vemancha"
```
Comment (jonthysell) [inline packages/sample-app-fabric/windows/SampleAppFabric.Package/Package.appxmanifest:11]: You'll want to remove your alias from here
```suggestion
    Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"
```

File: packages/sample-app-fabric/windows/SampleAppFabric.Package/Package.appxmanifest [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    Version="1.0.0.0" />
+
+  <Properties>
+    <DisplayName>SampleAppFabric</DisplayName>
+    <PublisherDisplayName>vemancha</PublisherDisplayName>
```
Comment (jonthysell) [inline packages/sample-app-fabric/windows/SampleAppFabric.Package/Package.appxmanifest:16]: You'll want to remove your alias here too

```suggestion
    <PublisherDisplayName>react-native-windows-testing</PublisherDisplayName>
```

File: packages/sample-app-fabric/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "@react-native/typescript-config/tsconfig.json"
```
Comment (jonthysell) [inline packages/sample-app-fabric/tsconfig.json:2]: This should match other apps in our mono repo.
```suggestion
  "extends": "@rnw-scripts/ts-config",
  "exclude": [
    "node_modules"
  ]
```

File: packages/sample-app-fabric/package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  "devDependencies": {
+    "@babel/core": "^7.20.0",
+    "@babel/generator": "^7.20.0",
+    "@babel/preset-env": "^7.20.0",
+    "@babel/preset-typescript": "^7.8.3",
+    "@babel/runtime": "^7.20.0",
+    "@react-native/babel-preset": "0.75.0-nightly-20240307-ff03b149e",
+    "@react-native/eslint-config": "0.75.0-nightly-20240307-ff03b149e",
+    "@react-native/metro-config": "0.75.0-nightly-20240307-ff03b149e",
+    "@react-native/typescript-config": "0.75.0-nightly-20240307-ff03b149e",
```
Comment (jonthysell) [inline packages/sample-app-fabric/package.json:28]: Remove this line.

```suggestion
```

File: packages/sample-app-fabric/Gemfile
Comment (jonthysell) [inline packages/sample-app-fabric/Gemfile:1]: Delete this file.

File: packages/sample-app-fabric/.watchmanconfig
Comment (jonthysell) [inline packages/sample-app-fabric/.watchmanconfig:1]: Delete this file.

File: packages/sample-app-fabric/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  root: true,
```
Comment (jonthysell) [inline packages/sample-app-fabric/.eslintrc.js:2]: Remove this line.

```suggestion
```

File: packages/sample-app-fabric/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+  extends: '@react-native',
```
Comment (jonthysell) [inline packages/sample-app-fabric/.eslintrc.js:3]:
```suggestion
  extends: ['@rnw-scripts'],
  parserOptions: { tsconfigRootDir: __dirname },
```

File: packages/sample-app-fabric/README.md
Comment (jonthysell) [inline packages/sample-app-fabric/README.md:1]: The README.me could probably be trimmed down to something like:

```
# sample-fabric-app

This package is used to verify a standalone fabric app.
```

File: packages/sample-app-fabric/.prettierrc.js
Comment (jonthysell) [inline packages/sample-app-fabric/.prettierrc.js:1]: Delete his file.

File: packages/sample-app-fabric/jest.config.js
Comment (jonthysell) [inline packages/sample-app-fabric/jest.config.js:1]: I would just replace this file with the contents of the `jest.config.windows.js` file.

File: packages/sample-app-fabric/jest.config.windows.js
Comment (jonthysell) [inline packages/sample-app-fabric/jest.config.windows.js:1]: Once you've copied the contents of this into the regular jest.config.js file, you should be able to delete this file.

File: packages/sample-app-fabric/package.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  "name": "sample-app-fabric",
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "start": "react-native start",
+    "lint": "rnw-scripts lint",
+    "lint:fix": "rnw-scripts lint:fix",
+    "watch": "rnw-scripts watch",
+    "windows": "react-native run-windows",
+    "test": "jest",
```
Comment (jonthysell) [inline packages/sample-app-fabric/package.json:11]: Update with this:

```suggestion
    "test": "rnw-scripts test"
```

File: packages/sample-app-fabric/package.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    "test:windows": "jest --config jest.config.windows.js"
```
Comment (jonthysell) [inline packages/sample-app-fabric/package.json:12]: This line can be removed.
```suggestion
```

File: packages/sample-app-fabric/package.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    "react-native": "0.75.0-nightly-20240307-ff03b149e",
+    "react-native-windows": "^0.0.0-canary.811"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.20.0",
+    "@babel/generator": "^7.20.0",
+    "@babel/preset-env": "^7.20.0",
+    "@babel/preset-typescript": "^7.8.3",
+    "@babel/runtime": "^7.20.0",
+    "@react-native/babel-preset": "0.75.0-nightly-20240307-ff03b149e",
```
Comment (jonthysell) [inline packages/sample-app-fabric/package.json:25]: Remove this line.
```suggestion
```

File: packages/sample-app-fabric/package.json [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    "@react-native/eslint-config": "0.75.0-nightly-20240307-ff03b149e",
```
Comment (jonthysell) [inline packages/sample-app-fabric/package.json:26]: Remove this line.
```suggestion
```

**Review by jonthysell — CHANGES_REQUESTED** [review]
Make sure you re-run `yarn install` from the repo root after modifying the package.json files.

File: packages/sample-app-fabric/package.json [hunk 11]
```diff
@@ -8,24 +8,22 @@
...
-    "test": "jest",
-    "test:windows": "jest --config jest.config.windows.js"
+    "test": "rnw-scripts test"
   },
   "dependencies": {
     "react": "18.2.0",
     "react-native": "0.75.0-nightly-20240307-ff03b149e",
-    "react-native-windows": "^0.0.0-canary.811"
+    "react-native-windows": "^0.0.0-canary.811",
+	"@typescript-eslint/eslint-plugin": "^5.21.0",
```
Comment (jonthysell) [inline packages/sample-app-fabric/package.json:17]:
```suggestion
```

File: packages/sample-app-fabric/windows/SampleAppFabric.Package/Package.appxmanifest [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="SampleAppFabric"
+    Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"
+    Version="1.0.0.0" />
+
+  <Properties>
+    <DisplayName>SampleAppFabric</DisplayName>
+    <PublisherDisplayName>vemancha</PublisherDisplayName>
```
Comment (jonthysell) [inline packages/sample-app-fabric/windows/SampleAppFabric.Package/Package.appxmanifest:16]:
```suggestion
    <PublisherDisplayName>react-native-windows-testing</PublisherDisplayName>
```

File: packages/sample-app-fabric/__tests__/App.test.tsx
Comment (jonthysell) [inline packages/sample-app-fabric/__tests__/App.test.tsx:1]: This file didn't need to be deleted. This is the test that should run when you run `yarn test`.

**Review by jonthysell — APPROVED** [review]
One you apply the changes an re-run yarn, and commit any lock file changes, it should be good to go.

**Review by marlenecota — APPROVED** [review]

**Pull request merged**